DataManagementSystem/DB/FileCatalogComponents/FileManager/FileManagerPs.py

```diff
@@ -775,51 +775,53 @@ def _setFileParameter(self, fileID, paramName, paramValue, connection=False):
   # _getFileReplicas related methods
   #
 
-  def _getFileReplicas(self, fileIDs, fields_input=['PFN'], allStatus=False, connection=False):
+  def _getFileReplicas(self, fileIDs, fields_input=None, allStatus=False, connection=False):
     """ Get replicas for the given list of files specified by their fileIDs
         :param fileIDs : list of file ids
-        :param fields_input : metadata of the Replicas we are interested in
+        :param fields_input : metadata of the Replicas we are interested in (default to PFN)
         :param allStatus : if True, all the Replica statuses will be considered,
                            otherwise, only the db.visibleReplicaStatus
 
         :returns S_OK with a dict { fileID : { SE name : dict of metadata } }
     """
 
-    connection = self._getConnection(connection)
+    if fields_input is None:
+      fields_input = ['PFN']
 
     fields = list(fields_input)
 
+    # always add Status in the list of required fields
     if 'Status' not in fields:
       fields.append('Status')
 
-    replicas = {}
+    # We initialize the dictionnary with empty dict
+    # as default value, because this is what we want for
+    # non existing replicas
+    # note (do NOT use dict.fromkeys, otherwise you get
+    # the same dict for all the fileIDs)
```

Comment:
Do we need comments like this? 
```suggestion
```
Also I think this belongs just before `for row in rows:` to make clear that is isn't actually used until then.

Reply:
What should go before `for row in rows` ? The `replicas` declaration or the comment ?

Reply:
The replicas declaration

Reply:
it's not used before the `for row` part, but I don't want it to be reinitialized for every chunk.
I can also remove the comment, but if anyone ever changes that to a `fromdict` (including me :-p ), you owe me a huge beer :-D

Reply:
Ah sorry I missed the chunked for loop 🤦